### PR TITLE
fix: pass label position flag to label rendering in BottomTabBar

### PR DESCRIPTION
**Pull request by EyMaddis**
In the process of upgrading from v4, I noticed a regression. 
In the past, the function form of `tabBarLabel` did get an `orientation: 'landscape' | 'portrait'`, this is no longer the case.
However, when using a custom Text rendering, we need to apply a margin to the text in horizontal mode.

Since the orientation/horizontal state is decided based on internal heuristics, It is a huge pain with a high bug potential when reimplementing that detection myself.

### CODE REVIEW

File: packages/bottom-tabs/src/types.tsx [hunk 1]
```diff
@@ -63,12 +63,16 @@ export type BottomTabNavigationOptions = {
...
+   * or a function that given { focused: boolean, color: string, orientation: 'horizontal' | 'vertical' } returns a React.Node to display in tab bar.
    * When undefined, scene title is used. To hide, see tabBarOptions.showLabel in the previous section.
    */
   tabBarLabel?:
     | string
-    | ((props: { focused: boolean; color: string }) => React.ReactNode);
+    | ((props: {
+        focused: boolean;
+        color: string;
+        orientation: 'horizontal' | 'vertical';
```
Comment (satya164) [inline packages/bottom-tabs/src/types.tsx:74]:
```suggestion
        position: LabelPosition;
```

File: packages/bottom-tabs/src/views/BottomTabItem.tsx [hunk 2]
```diff
@@ -30,7 +30,11 @@ type Props = {
    */
   label:
     | string
-    | ((props: { focused: boolean; color: string }) => React.ReactNode);
+    | ((props: {
+        focused: boolean;
+        color: string;
+        orientation: 'horizontal' | 'vertical';
```
Comment (satya164) [inline packages/bottom-tabs/src/views/BottomTabItem.tsx:36]:
```suggestion
        position: LabelPosition;
```

File: packages/bottom-tabs/src/views/BottomTabItem.tsx [hunk 3]
```diff
@@ -206,7 +210,11 @@ export default function BottomTabBarItem({
       );
     }
 
-    return label({ focused, color });
+    return label({
+      focused,
+      color,
+      orientation: horizontal ? 'horizontal' : 'vertical',
```
Comment (satya164) [inline packages/bottom-tabs/src/views/BottomTabItem.tsx:216]:
```suggestion
      position: horizontal ? 'beside-icon' : 'below-icon',
```

File: packages/bottom-tabs/src/views/BottomTabItem.tsx (hunk 2 above)
Comment (satya164) [reply packages/bottom-tabs/src/views/BottomTabItem.tsx:36]: Also need to add the import for `LabelPosition` above

**Comment by EyMaddis** [comment]
@satya164 thanks for the feedback, I added the changes and improved the documentation a bit.

**Pull request merged**